quantum/process_keycode/process_dynamic_macro.c

```diff
@@ -205,6 +205,12 @@ bool process_dynamic_macro(uint16_t keycode, keyrecord_t *record) {
                     dynamic_macro_record_start(&macro_pointer, r_macro_buffer);
                     macro_id = 2;
                     return false;
+#ifdef DYNAMIC_MACRO_NO_NESTING
```

Comment:
So this block changes DYN_MACRO_PLAY1/2 from keydown to keyup? Does that matter in the NO_NESTING case?

Reply:
It gave no issues in my testing, which was with an otherwise vanilla setup (i.e. no handlers in process_record_user regarding the DYN_MACRO keys). I don't know if there is some user code that exactly relies on the previous behaviour though!

Reply:
Could you test with the original block of code, as i cant see why this part of the change is required.

Reply:
If you mean to test it without changing the play behaviour from keyup to keydown (I just notice you had it the wrong way around in your earlier comment):
I have tested it without changing it to keyup. The result was that the keystroke that (now) stops the recording (on keydown), will also cause the replay of the dynamic macro immediately on keyup (as that's how it was implemented before). I don't think that's desired behaviour.